### DISCUSSION TITLE
Docker setup and port issues

### DISCUSSION THREAD

**Discussion by maste9** [music-assistant/support#2056] [Q&A] [answered]
**_tl;dr:_ The host mode doesn't seem to work with MA *docker setups* with *slimproto* anymore. Need to use ports *3483*, *8095* and *8097* in docker-compose.yml instead.**

As I had some struggle with my MA/HA docker-based setup after the last MA updates I want to share the results/solution I found getting MA to work with my **squeezelite** players again.

After the latest update my MA complained about port 3483 was already taken. This was consistently reproducible by turning the slimproto protocol on and off in the settings.
I thoroughly checked for any containers or apps that could use that port (suddenly) but couldn't find.

Then I changed my docker-compose.yml from host-mode to ports like the following
```
#    network_mode: host
#    privileged: true
    ports:
      - 3483:3483
      - 8095:8095
```

Now my setup found the player again - and got control about the 3483 port but unfortunately couldn't  play any songs. The logs showed that MA lacked access to the files to play.

So lately I realized that I also needed to open port 8097 for the UI to communicate with and after adding that to the docker-compose.yml ...

```
#    network_mode: host
#    privileged: true
    ports:
      - 3483:3483
      - 8095:8095
      - 8097:8097
```
... MA is working again like before.

I'm still confused why host mode isn't working any longer or which container my interfere with the ports. But at least it works now again :slightly_smiling_face: 

Hope that might help other folks.

**Comment by OzGav** [comment]
I am not sure why host network mode isn’t working for you. This is what should be used due to the large number of ports that MA can potentially require.

**Comment by marcelveldt** [comment]
This is NOT a solution, you need host network.

I think you have another slimproto server running, such as Logitech Media Server or the slimproto integration in HA

**Reply by maste9** [reply]
Hmm ... Ok, I was afraid that it won't be so easy :confused: 

What's puzzling me is: How could the port 3483 work with ports and be blocked with host? I mean ... shouldn't docker either block in both cases or none? :thinking: 

And I actually tried to really make sure that there's no LMS or slimpro running in my HA container.
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/4653372/d6878f08-d4d6-4242-98ba-305a1a47b3ef)
Using only the MA integration from HACS - else no integrations which could use that port, do they?

**Reply by maste9** [reply, marked as answer]
Well ... that's interesting:

I gave it another shot now, switched it back to host - and again it didn't work. But the logs changed. The error message didn't longer complain about port 3483 but about "any of the ports" regarding slimproto. It suddenly mentioned port 9000. Which was already used by portainer. After I changed the port of portainer to 9001 MA worked with host again ....

So I guess the 9000 port was the reason for the blockage before and the message about the 3483 port was just misleading before ...
Would that be a possible explanation?

**Comment by adrianipopescu** [comment]
@maste9 managed to figure this one out without host mode?

**Reply by maste9** [reply]
No, I switched everything to host mode as the devs [declared that necessary](https://github.com/orgs/music-assistant/discussions/2056#discussioncomment-8898096).

**Comment by aaron-rana** [comment]
in my case cockpit was using 9090. just fyi
